File: app/src/main/java/com/kxjsj/doctorassistant/Appxx/Sicker/KnowledgeF.java
package com.kxjsj.doctorassistant.Appxx.Sicker;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.kxjsj.doctorassistant.App;
import com.kxjsj.doctorassistant.Appxx.Sicker.Hospital.IDInfoActivity;
import com.kxjsj.doctorassistant.Appxx.Sicker.Hospital.SelfPayActivity;
import com.kxjsj.doctorassistant.Component.BaseFragment;
import com.kxjsj.doctorassistant.JavaBean.PatientBed;
import com.kxjsj.doctorassistant.Net.ApiController;
import com.kxjsj.doctorassistant.R;
import com.kxjsj.doctorassistant.Rx.DataObserver;
import com.kxjsj.doctorassistant.Utils.K2JUtils;

import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

/**
 * Created by vange on 2017/9/28.
 */

public class KnowledgeF extends BaseFragment {


    private Unbinder unbinder;
    private PatientBed beans;

    @Override
    protected void initView(@Nullable Bundle savedInstanceState) {
        setRetainInstance(true);
        if (savedInstanceState == null) {
            loadLazy();
        }else{
         beans= (PatientBed) savedInstanceState.getSerializable("beans");
        }
    }

    @Override
    protected int getLayoutId() {
        return R.layout.find_infos;
    }


    @Override
    protected void loadLazy() {
        ApiController.getBedInfo(
                App.getUserInfo().getPatientNo())
                .subscribe(new DataObserver<PatientBed>(this) {
                    @Override
                    public void OnNEXT(PatientBed bean) {
                        if(bean==null){
                            K2JUtils.toast("获取信息失败");
                        }
                        beans =bean;
                    }

                    @Override
                    public void onError(Throwable e) {
                        super.onError(e);
                    }
                });
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // TODO: inflate a fragment view
        View rootView = super.onCreateView(inflater, container, savedInstanceState);
        unbinder = ButterKnife.bind(this, rootView);
        return rootView;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putSerializable("beans",beans);
    }
    @OnClick({R.id.id, R.id.checkinfo, R.id.medicalinfo, R.id.money, R.id.checke_price, R.id.roominfo})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.id:
//                if(beans==null)
//                    return;
                Intent intent = new Intent(getContext(), IDInfoActivity.class);
                intent.putExtra("bean", beans);
                startActivity(intent);
                break;
            case R.id.checkinfo:
                break;
            case R.id.medicalinfo:
                break;
            case R.id.money:
                startActivity(new Intent(getContext(), SelfPayActivity.class));
                break;
            case R.id.checke_price:
                break;
            case R.id.roominfo:
                break;
        }
    }


}

File: app/src/main/java/com/kxjsj/doctorassistant/Appxx/Sicker/KnowledgeF.kt
package com.kxjsj.doctorassistant.Appxx.Sicker

import android.content.Context
import android.os.Bundle
import android.view.View
import com.kxjsj.doctorassistant.Appxx.Sicker.Knowledge.knowledgePagerAdapter
import com.kxjsj.doctorassistant.Component.BaseFragment
import com.kxjsj.doctorassistant.JavaBean.KotlinBean
import com.kxjsj.doctorassistant.Net.ApiController
import com.kxjsj.doctorassistant.R
import com.kxjsj.doctorassistant.Rx.DataObserver
import com.nestrefreshlib.State.DefaultStateListener
import com.nestrefreshlib.State.StateLayout
import kotlinx.android.synthetic.main.knowledge_layout.*

/**
 * Created by vange on 2018/1/10.
 */
class KnowledgeF : BaseFragment() {
    var stateLayout: StateLayout? = null
    var adapter: knowledgePagerAdapter? = null
    var beanx: ArrayList<KotlinBean.Title>? = null
    override fun initView(savedInstanceState: Bundle?) {
        retainInstance = true
        val serializable = savedInstanceState?.getSerializable("beanx")
        if (serializable != null)
            beanx = serializable as ArrayList<KotlinBean.Title>?
        if (beanx != null) {
            initAdapter(beanx)
        }
        if(beanx==null)
        getData()
    }

    override fun getLayoutId(): Int {

        return R.layout.knowledge_layout
    }

    private fun getData() {
        ApiController.getknowledgeBaseType()
                .subscribe(object : DataObserver<ArrayList<KotlinBean.Title>>(context) {
                    override fun OnNEXT(beans: ArrayList<KotlinBean.Title>?) {
                        if (beans?.size == 0)
                            stateLayout?.showEmpty()
                        else
                            stateLayout?.showItem()
                        if (beanx == null) {
                            beanx = beans
                        } else {
                            beanx?.addAll(beans!!)
                        }
                        initAdapter(beanx)
                    }

                    override fun OnERROR(error: String?) {
                        super.OnERROR(error)
                        stateLayout?.ShowError()
                    }
                })

    }

    private fun initAdapter(beans: ArrayList<KotlinBean.Title>?) {
        adapter = knowledgePagerAdapter(childFragmentManager, beans!!)
        adapter?.datas=beans!!
        viewpager.adapter = adapter
        tablayout.setupWithViewPager(viewpager)
    }

    override fun SetRootView(): View? {
        if (view != null)
            return view
        stateLayout = StateLayout(context)
        return stateLayout
                ?.setContent(layoutId)
                ?.setStateListener(object : DefaultStateListener() {
                    override fun netError(p0: Context?) {
                        getData()
                    }
                })
    }

    override fun loadLazy() {

    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        outState?.putSerializable("beanx", beanx)
    }
}